pagopa/hub-spid-login-ms
# [#HSL-4] E2E tests for access logs on Azure Storage

This PR introduces a test scenario that does the following:
* Start the application with an Azure Storage account attached and `ENABLE_SPID_ACCESS_LOGS=true`
* Run a successful login
* Query storage to check if a log is present

Request content isn't inspected

### Caveats
* From an E2E perspective, we can't relate a log to a specific login session; thus, tests are dummy and only check for log presence. For finer-grained tests, the check must be improved. 

### Changes
See commit list

### ⚠️ Tests aren't currently working due to issues with Azurite. 
The following error arise at writing time. The error is not present when tests are executed against a real instance of Azure Storage
```
StorageError: Server failed to authenticate the request. Make sure the value of the Authorization header is formed correctly including the signature
```
